src/pyaro/timeseries/Filter.py

```diff
@@ -594,15 +636,15 @@ def _index_from_include_exclude(self, times1, times2, includes, excludes):
 
         return idx
 
-    def has_envelope(self):
+    def has_envelope(self) -> bool:
         """Check if this filter has an envelope, i.e. a earliest and latest time"""
         return (
             len(self._start_include)
             or len(self._startend_include)
             or len(self._end_include)
         )
 
-    def envelope(self) -> tuple[datetime, datetime]:
+    def envelope(self) -> TimeBound:
```

Comment:
I like TimeBound as general input-type, but the output-type of envelope should stay a datetime. (Unnecessary to burden the user with handling with 3 types. Unnecessary to change signature from datetime to np.datetime64 here.)